Sources/StreamChatSwiftUI/ChatChannel/MessageList/Polls/PollAttachmentView.swift

```diff
@@ -0,0 +1,271 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import SwiftUI
+
+public struct PollAttachmentView<Factory: ViewFactory>: View {
+    
+    @Injected(\.chatClient) var chatClient
+    @Injected(\.fonts) var fonts
+    @Injected(\.colors) var colors
+    
+    private let factory: Factory
+    private let message: ChatMessage
+    private let isFirst: Bool
+    
+    @StateObject var viewModel: PollAttachmentViewModel
+    
+    public init(
+        factory: Factory,
+        message: ChatMessage,
+        poll: Poll,
+        isFirst: Bool
+    ) {
+        self.factory = factory
+        self.message = message
+        self.isFirst = isFirst
+        _viewModel = StateObject(
+            wrappedValue: PollAttachmentViewModel(
+                message: message,
+                poll: poll
+            )
+        )
+    }
+    
+    public var body: some View {
+        VStack(spacing: 16) {
+            VStack(alignment: .leading, spacing: 2) {
+                HStack {
+                    Text(poll.name)
+                        .font(fonts.bodyBold)
+                    Spacer()
+                }
+                
+                HStack {
+                    Text(subtitleText)
+                        .font(fonts.caption1)
+                        .foregroundColor(Color(colors.textLowEmphasis))
+                    Spacer()
+                }
+            }
+            
+            ForEach(options.prefix(10)) { option in
```

Comment:
🤔  10 could be a constant since it is used in multiple places and has potential of being configurable value

Reply:
sure, feel free to add it